staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -83,3 +95,30 @@ func (record *attributesRecord) GetOldObject() runtime.Object {
 func (record *attributesRecord) GetUserInfo() user.Info {
 	return record.userInfo
 }
+
+func (record *attributesRecord) SetAnnotations(pluginName string, annotations map[string]string) {
+	if record.annotations == nil {
+		return
```

Comment:
Is it a conscious decision to silently discard annotation instead of creating a new map? Could you please link me to a discussion?

Reply:
https://github.com/kubernetes/kubernetes/pull/58679#discussion_r168297676

Reply:
Why not always initialize this map to non-nil value?

Reply:
> Why not always initialize this map to non-nil value?

The reason is explained here:
https://github.com/kubernetes/kubernetes/pull/58679#discussion_r168297676